substrate/frame/revive/rpc/src/tests.rs

```diff
@@ -275,3 +279,133 @@ async fn invalid_transaction() -> anyhow::Result<()> {
 
 	Ok(())
 }
+
+async fn get_evm_block_from_storage(
+	node_client: &OnlineClient<SrcChainConfig>,
+	block_number: U256,
+) -> anyhow::Result<Block> {
+	let mut blocks = node_client.blocks().subscribe_best().await?;
+
+	// Get current best block
+	let block_hash = if let Some(Ok(block)) = blocks.next().await {
+		if block_number == block.number().into() {
+			block.hash()
+		} else {
+			return Err(anyhow::anyhow!(
+				"Block number mismatch found: {:?} required: {block_number:?}",
+				block.hash()
+			));
+		}
+	} else {
+		return Err(anyhow::anyhow!("Failed to fetch next block"));
+	};
```

Comment:
does subscribe_best()  block until we get the next block?

if so it will block until the next block is mined
if not isn't there a better API than subscribe_best().next() ?